app/models/node.rb

```diff
@@ -73,6 +73,11 @@ def options=(value)
     super(value.to_h)
   end
 
+  def should_generate_new_friendly_id?
```

Comment:
By the friendly_id magic. See http://norman.github.io/friendly_id/file.Guide.html#Deciding_When_to_Generate_New_Slugs if you want to look it up.
Actually, I could add that link as a comment if you think it will be useful.